extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/transferhistory/TransferHistoryPageApiService.java

```diff
@@ -14,14 +14,112 @@
 
 package de.sovity.edc.ext.wrapper.api.ui.pages.transferhistory;
 
-import de.sovity.edc.ext.wrapper.api.ui.model.TransferHistoryPage;
+import de.sovity.edc.ext.wrapper.api.ui.model.ContractAgreementDirection;
+import de.sovity.edc.ext.wrapper.api.ui.model.TransferHistoryEntry;
+import de.sovity.edc.ext.wrapper.api.ui.pages.contracts.services.TransferProcessStateService;
 import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.connector.contract.spi.negotiation.store.ContractNegotiationStore;
+import org.eclipse.edc.connector.contract.spi.types.agreement.ContractAgreement;
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiation;
+import org.eclipse.edc.connector.spi.asset.AssetService;
+import org.eclipse.edc.connector.spi.contractagreement.ContractAgreementService;
+import org.eclipse.edc.connector.spi.transferprocess.TransferProcessService;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcess;
+import org.eclipse.edc.spi.entity.Entity;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.edc.spi.types.domain.asset.Asset;
+import org.jetbrains.annotations.NotNull;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import static de.sovity.edc.ext.wrapper.utils.EdcDateUtils.utcMillisToOffsetDateTime;
+import static java.util.stream.Collectors.*;
 
 @RequiredArgsConstructor
 public class TransferHistoryPageApiService {
 
-    public TransferHistoryPage transferHistoryPage() {
-        // TODO: Implement
-        throw new IllegalStateException("Not yet implemented");
+    private final AssetService assetService;
+    private final ContractAgreementService contractAgreementService;
+    private final ContractNegotiationStore contractNegotiationStore;
+    private final TransferProcessService transferProcessService;
+    private final TransferProcessStateService transferProcessStateService;
+
+    /**
+     * Fetches all Transfer History entries as {@link TransferHistoryEntry}s.
+     *
+     * @return {@link TransferHistoryEntry}s
+     */
+    @NotNull
+    public List<TransferHistoryEntry> getTransferHistoryEntries() {
+
+        var negotiationsById = getAllContractNegotiations().stream()
+                .filter(it -> it.getContractAgreement() != null)
```

Comment:
rename `it` to `negotiation`. This enables the reader to directly see what is beeing used instead of inferring it from the context